### DISCUSSION TITLE
[Specification] Font awesome and others fonts should be built in.

### DISCUSSION THREAD

**Discussion by davidbuckleyni** [dotnet/maui#312] [Ideas]
# [The feature] 

Fontawsome is that much part of mobile apps as it is web apps and should be able to be used inside a Maui Phone app as default.
# API

For Example a button could have 

    <button x:name="btnPlanes" Icon="fa fa-plane fa-xs" IconPosition="Bottom"  Text="Plane"  TextPosition=TextPositionEnum font-Icon="fa-plane" />

When a use would enter the words fa  a list of icons would be display in a dropdown editor element much the same way colours do. Then the user could pick the icon and would auto fill the Icon values their would also be a

    IconPositionEnum=Bottom, Top,Left,Right, Center then could be a use case where one would need Bottom+Centre
IconPoistion 

Size would be controlled in the same way it does at present in the web fonts in that you would have the attribute fa-xs for example would make a icon xtra small size  spec here. https://fontawesome.com/how-to-use/on-the-web/styling/sizing-icons 
When a user does x2 x4 it should make the icon the appropriate sizes

It should be that we could use font awsome icons and icon-icons another popular font icon. 

It just feels that there is allot of configuration in downloading the fonts etc just to get them working.

**Comment by REscobar** [comment]
I believe the framework should be unopinionated about which font you should use as an icon font, but should make it easier to use any font glyph as an icon or image source

**Comment by nextcodelab** [comment]
xaml path, svg, image, fontIcons... we can use all of this is the best.. like this app has thousands of xaml path, svg path and android vector xml path....
[https://www.microsoft.com/en-us/p/xaml-path-icons/9mtbnqsz9nz9?activetab=pivot:overviewtab](https://www.microsoft.com/en-us/p/xaml-path-icons/9mtbnqsz9nz9?activetab=pivot:overviewtab)


https://user-images.githubusercontent.com/73583588/103608181-b0165480-4f55-11eb-8ec3-58bfadc7bcc2.mp4

**Comment by REscobar** [comment]
You may want to disclose who the developer is

**Comment by bondarenkod** [comment]
You can build a NuGet package with this or any other font and start using it even with XF.
There is no single reason to include (embed) any fonts by default.

**Comment by mrlacey** [comment]
If you want a way to configure the inclusion of FontAwesome in a new app, you can do it in one click with  [Maui App Accelerator](https://marketplace.visualstudio.com/items?itemName=MattLaceyLtd.MauiAppAccelerator)
